lib/puppet/provider/user/useradd.rb

```diff
@@ -203,7 +203,10 @@ def deletecmd
     else
       cmd = [command(:delete)]
     end
-    cmd += @resource.managehome? ? ['-r'] : []
+    # Solaris `useradd -r` will fail if the homedir does not exist.
```

Comment:
Looks like not every platform requires the homedir exist. From Solaris' man page
```
     -r               Remove the user's home directory  from  the
                      system.  This  directory  must  exist.  The
                      files and directories under the home direc-
                      tory will no longer be accessible following
                      successful execution of the command. A  ZFS
                      dataset  that  was  created  for the user's
                      home  directory   will   be   removed.   An
                      auto_home entry that was added for the user
                      will be deleted.
```